src/Microsoft.OpenApi/Extensions/OpenApiSerializableExtensions.cs

```diff
@@ -114,6 +114,42 @@ public static void Serialize<T>(this T element, IOpenApiWriter writer, OpenApiSp
             writer.Flush();
         }
 
+        /// <summary>
+        /// Serializes the <see cref="IOpenApiSerializable"/> to Open API document using the given specification version and writer.
+        /// </summary>
+        /// <typeparam name="T">the <see cref="IOpenApiSerializable"/></typeparam>
+        /// <param name="element">The Open API element.</param>
+        /// <param name="writer">The output writer.</param>
+        public static void Serialize<T>(this T element, IOpenApiWriter writer)
+            where T : IOpenApiSerializable
+        {
+            if (element == null)
+            {
+                throw Error.ArgumentNull(nameof(element));
+            }
+
+            if (writer == null)
+            {
+                throw Error.ArgumentNull(nameof(writer));
+            }
+
+            switch (writer.Settings.SpecVersion)
+            {
+                case OpenApiSpecVersion.OpenApi3_0:
+                    element.SerializeAsV3(writer);
+                    break;
+
+                case OpenApiSpecVersion.OpenApi2_0:
+                    element.SerializeAsV2(writer);
+                    break;
+
+                default:
+                    throw new OpenApiException(string.Format(SRResource.OpenApiSpecVersionNotSupported, writer.Settings.SpecVersion));
+            }
+
+            writer.Flush();
```

Comment:
Can we leverage the method above?


return element.Serialize(writer, writer.Settings.SpecVersion);

Reply:
Fixed. Thanks!

---
In reply to: [193587359](https://github.com/Microsoft/OpenAPI.NET/pull/267#discussion_r193587359) [](ancestors = 193587359)